Bump GitHub actions to latest versions

deining opened the pull request:
This PR bumps GitHub workflow actions to their latest versions. It also fixes warnings as e.g. seen [here](https://github.com/pdfcpu/pdfcpu/actions/runs/12121154268).

```
Restore cache failed: Dependencies file is not found in /home/runner/work/pdfcpu/pdfcpu.
Supported file pattern: go.sum
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/pdfcpu/pdfcpu?pullRequest=1028) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/pdfcpu/pdfcpu?pullRequest=1028) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/pdfcpu/pdfcpu?pullRequest=1028) it.</sub>

fancycode commented:
Might be a good idea to setup @dependabot to handle future dependency updates automatically.

hhrutter commented:
Excellent idea.

deining commented:
> Excellent idea.

Just added a commit that implements automatic update checking via dependabot.

hhrutter commented:
I would like the update interval to be weekly without a specific time.
If you also leave test.yml untouched I am happy to accept this PR.
The dependabot PR will take care of the rest.

deining commented:
> I would like the update interval to be weekly without a specific time.

I just changed `dependabot.yml` as requested.

> If you also leave test.yml untouched

Reverted `test.yml`, but swapped checkout and go setup action in order to avoid the warning `Restore cache failed`.

> I am happy to accept this PR.

Thanks for merging!

hhrutter reviewed (commented):
Go modules update interval is still daily.

deining commented:
> Go modules update interval is still daily.

Oops, sorry. This is fixed now.

hhrutter commented:
Thanks!
